### DISCUSSION TITLE
Feedback on Task Lists

### DISCUSSION THREAD

**Discussion by Danielku15** [community/community#5238] [Projects and Issues]
I really like the task lists to make interactive checkboxes. But often I also just want to have interactive checkboxes in my issue/PR templates and they do not represent actual tasks. 

I somehow "misuse" them currently as interactive template for people contributing so that they can fill out some details like here: 

![image](https://user-images.githubusercontent.com/674916/130352782-b25993ba-5955-469f-b8ea-3025e04f46c2.png)

Not all checkmarks are really mandatory and they are not all tasks. This leads to a display like: 
![image](https://user-images.githubusercontent.com/674916/130352803-786636f1-f9b9-4105-b436-c23e2df9240a.png)

I have seen some very early access feature of "issue form templates" which really allow you to make forms with UI components to enforce a certain use input structure. This feature might also solve this at some point if they become available for PRs too. But until then it would be great if we can somehow indicate which checkmarks are actually tasks and which ones are just checkboxes. Maybe using a different syntax like: 

```
- { } Unchecked
- {x} Checked
```
 
Parenthesis would be too close to the looks of radio buttons, but braces could work to have a checkbox which is not a task.

**Comment by Zadjil-AR** [comment]
I really like this suggestion and was going to add it myself. Others have also done so in #5214  and #5197. 
We use the checkboxes for a similar purpose - as flags that can be marked off to provide information about the PR or issue.
These are then processed using a bot to get information about what has been submitted.

I also like the syntax that you propose.

**Comment by nednednerb** [comment]
Our organization uses the checkboxes in this manner too.

**Comment by Kixunil** [comment]
> very early access feature of "issue form templates" which really allow you to make forms with UI components to enforce a certain use input structure

OT, but I really, really want to thank you for mentioning it! Didn't know about it, this will help me a lot! For those who want to try as well check [the docs](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms)

**Comment by ciriarte** [comment]
Same idea here. We use checkboxes for other purposes (like gathering metadata about the issue or pull request.)

**Comment by KarimElsayad247** [comment]
Second this.

it allows us to specify what components of our system an issue should affect, it is **not** tasks, so it's misleading. This seems to be a popular feature request, so i hope github eventually implements it.

**Comment by xmo-odoo** [comment]
> braces could work to have a checkbox which is not a task.

FWIW I've also seen markdown extensions for radio buttons (the ones I've seen used parens), which would also be useful as commonly the tasks / pseudo-forms give multiple choices only one of which makes sense.

Sometimes a template picker step is sufficient, but not always.

**Comment by JaffaKetchup** [comment]
Anyone from GitHub willing to do anything about this? I have the same problem as all the others above.

**Comment by Malix-Labs** [comment]
#19502

**Comment by omerbensaadon** [comment]
Hey folks! So good news, the next generation of Task lists should resolve this friction. 

I am the PM working on it, we don't have a set time for a release yet. It's part of a larger effort. 

Rest assured, we hear you!

**Reply by Malix-Labs** [reply]
Hello @omerbensaadon,
Thanks for your answer to this problem!
Do we have more documentation about the

> next generation of task list

that you are talking about?
Best regards

**Reply by omerbensaadon** [reply]
Not quite yet sorry to say

**Reply by Malix-Labs** [reply]
Ok, no problem.
Is there any ETA?

**Reply by vilkinsons** [reply]
👀

**Reply by diegoholiveira** [reply]
Any news about the tasks list next gen?

**Reply by omerbensaadon** [reply]
Ah thank you all for the ping! Sign up for the Private Beta here: https://docs.github.com/en/issues/tracking-your-work-with-issues/about-tasklists

**Reply by nexushoratio** [reply]
My understanding is the new tasklists feature is on hold.  Is that correct?

**Comment by thejeff77** [comment]
Would be really great to have a task list button that can toggle between 3 states. 

1. Not Done (unchecked)
2. Done (checked)
3. Not Applicable (whole line crossed out)

Currently to make it crossed out requires some tedious manual editing of the pull request template..

**Reply by AlanSl** [reply]
This would be great, and could even just use the common [checkbox "indeterminate" state (like `[-]`)](https://github.com/community/community/discussions/19199); the crossing out of text, when applicable, could be done with normal strikethrough markup of the text.

There are lots of common GitHub scenarios where a task in a list is no longer to-do, but is not "done" in the current issue or PR either: e.g. was split into its own issue, done in a different PR, proved to be impossible or longer an issue, confirmed wontfix, etc etc.

**Comment by omerbensaadon** [comment]
There is not, but it’s actively being worked!

**Reply by Superskyyy** [reply]
Hi! Is there any update on this feature? :)

**Reply by omerbensaadon** [reply]
https://docs.github.com/en/issues/tracking-your-work-with-issues/about-tasklists

We're going to release this feature in the next few months which should resolve this friction 👍🏼 HTML checkboxes will no longer count as tasks once tasklists are released

**Reply by esafak** [reply]
Can we get an update @omerbensaadon ?

**Comment by anthonyalayo** [comment]
Can we get an update on this?

**Reply by omerbensaadon** [reply]
https://docs.github.com/en/issues/tracking-your-work-with-issues/about-tasklists

We're going to release this feature in the next few months which should resolve this friction 👍🏼 HTML checkboxes will no longer count as tasks once tasklists are released

**Reply by Zadjil-AR** [reply]
@omerbensaadon Thank you for the prompt update.
I already really like the new tasklists, and even more now that it is opening the ability to have non-task checkboxes.

Can you clarify if existing PRs & issues that have the old format of checkboxes will continue to list them as tasks?
I want to know if we should go back to currently open items and migrate the task lists to the new format to keep them as tasks when it comes out of beta.
If the existing items remain as task lists will there be a way to get the new behavior?

**Reply by omerbensaadon** [reply]
Once we GA tasklists (down the road) checkboxes will no longer result in completion counts

**Comment by monica-m-ps** [comment]
Hello. Is this fixed yet?

**Comment by Futuza** [comment]
I assume that the tasklist feature was disabled while you guys worked on this? But its nearly been nearly two years without an update...

Discussions like [these](https://github.com/orgs/community/discussions/149409) or [these](https://github.com/orgs/community/discussions/147675) also remain unanswered.  I miss the progress bar tasklists, please bring back/fix?
